File: frontend/src/services/notificationService.js
import api from './api';

const notificationService = {
    getUserNotifications: async (userId) => {
        const response = await api.get(`/notifications/user/${userId}`);
        return response.data;
    },
    markAsRead: async (notificationId) => {
        const response = await api.put(`/notifications/${notificationId}/read`);
        return response.data;
    },
    markAllAsRead: async (userId) => {
        const response = await api.put(`/notifications/user/${userId}/read-all`);
        return response.data;
    },
    // --- NEW: DELETE NOTIFICATION ---
    deleteNotification: async (notificationId) => {
        const response = await api.delete(`/notifications/${notificationId}`);
        return response.data;
    }
};

export default notificationService;

File: backend/src/main/java/com/sliit/smartcampus/modules/notification/service/NotificationService.java
package com.sliit.smartcampus.modules.notification.service;

import com.sliit.smartcampus.modules.notification.model.Notification;
import com.sliit.smartcampus.modules.notification.repository.NotificationRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class NotificationService {

    private final NotificationRepository notificationRepository;

    // --- UPDATED CREATE LOGIC ---
    public Notification createNotification(String userId, String title, String message, String category, String referenceId) {
        Notification notification = new Notification();
        notification.setUserId(userId);
        notification.setTitle(title);
        notification.setMessage(message);
        notification.setCategory(category != null ? category : "GENERAL");
        notification.setReferenceId(referenceId);
        return notificationRepository.save(notification);
    }

    public List<Notification> getUserNotifications(String userId) {
        return notificationRepository.findByUserIdOrderByCreatedAtDesc(userId);
    }

    public void markAsRead(String notificationId) {
        notificationRepository.findById(notificationId).ifPresent(notification -> {
            notification.setRead(true);
            notificationRepository.save(notification);
        });
    }

    public void markAllAsRead(String userId) {
        List<Notification> unread = notificationRepository.findByUserIdOrderByCreatedAtDesc(userId)
                .stream().filter(n -> !n.isRead()).toList();
        
        unread.forEach(n -> n.setRead(true));
        notificationRepository.saveAll(unread);
    }

    // --- NEW: DELETE LOGIC ---
    public void deleteNotification(String notificationId) {
        notificationRepository.deleteById(notificationId);
    }
}